Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,398 @@
+                return;
+            }
+
+            // Resolve path to user folder
+            AZ::IO::FixedMaxPath editorBookmarkFilePath =
+                AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Registry/ViewBookmarks/";
+
+            editorBookmarkFilePath /= m_bookmarkfileName;
+
+            AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+            dumperSettings.m_prettifyOutput = true;
+
+            AZStd::string bookmarkKey = "/" + m_bookmarkfileName;
+            dumperSettings.m_includeFilter = [&bookmarkKey](AZStd::string_view path)
+            {
+                AZStd::string_view o3dePrefixPath(bookmarkKey);
+                return o3dePrefixPath.starts_with(path.substr(0, o3dePrefixPath.size()));
+            };
+
+            AZStd::string stringBuffer;
+            AZ::IO::ByteContainerStream stringStream(&stringBuffer);
+            if (!AZ::SettingsRegistryMergeUtils::DumpSettingsRegistryToStream(*registry, "", stringStream, dumperSettings))
+            {
+                AZ_Warning(
+                    "SEditorSettings", false, R"(Unable to save changes to the Editor Preferences registry file at "%s"\n)",
+                    editorBookmarkFilePath.c_str());
+                return;
+            }
+
+            bool saved{};
+            constexpr auto configurationMode =
+                AZ::IO::SystemFile::SF_OPEN_CREATE | AZ::IO::SystemFile::SF_OPEN_CREATE_PATH | AZ::IO::SystemFile::SF_OPEN_WRITE_ONLY;
+            if (AZ::IO::SystemFile outputFile; outputFile.Open(editorBookmarkFilePath.c_str(), configurationMode))
+            {
+                saved = outputFile.Write(stringBuffer.data(), stringBuffer.size()) == stringBuffer.size();
+            }
+
+            // Once written to the desired file remove the key from the settings registry
+            registry->Remove(bookmarkKey + "/");
+            AZ_Warning(
+                "SEditorSettings", saved, R"(Unable to save Editor Preferences registry file to path "%s"\n)",
+                editorBookmarkFilePath.c_str());
+        }
+
+        bool ViewBookmarkLoader::SaveBookmark(ViewBookmark bookmark, StorageMode mode)
+        {
+            switch (mode)
+            {
+            case AzToolsFramework::ViewBookmarkLoaderInterface::Shared:
+                return SaveSharedBookmark(bookmark);
+            case AzToolsFramework::ViewBookmarkLoaderInterface::Local:
+                return SaveLocalBookmark(bookmark);
+            case AzToolsFramework::ViewBookmarkLoaderInterface::Invalid:
+                return false;
+            }
+            return false;
+        }
+
+        bool ViewBookmarkLoader::SaveLastKnownLocationInLevel(ViewBookmark bookmark)
+        {
+            return SaveLocalBookmark(bookmark, true);
+        }
+
+        bool ViewBookmarkLoader::LoadViewBookmarks()
+        {
+            struct ViewBookmarkVisitor : AZ::SettingsRegistryInterface::Visitor
```

Comment:
This can use either the the `AZ::SettingsRegistryVisitUtils::VisitObject` function to visit the direct children of the `/O3DE/ViewBookmarks` key or it can inherit from `AZ::SettingsRegistryVisitUtils::ObjectVisitor` to support the same visitation of direct children.

The benefits of using the SettingsRegistryVisitUtils, is that it will visit each of the direct child fields of the specified JSON path.
it will NOT recurse into children. That allows the user to choose to invoke a Visit function again to visit any descendant fields.

Reply:
Unrelated, but I will keep this in mind for other uses very soon :D